docs/develop/java/kafka.mdx

```diff
@@ -2,71 +2,30 @@
-* Wire up an event handler with a Kafka topic through a subscription
-
-## Event handler
-
-An event handlers is alike any other handler in your services/virtual objects, and profit from the same features, including durable execution, [service communication](/develop/java/service-communication) and [side effects](/develop/java/side-effects).
 
-**Handling keyed events within virtual objects**
+import {SubtleStep} from "../../../src/components/Stepper";
 
-You can handle events within virtual objects, where the key of the Kafka record matches the object key.
-
-Event handlers are no different from regular handlers, and the event payload will be (de)serialized as JSON:
-
-<Tabs groupId="sdk" queryString>
-<TabItem value="java" label="Java">
-
-```java
-CODE_LOAD::java/src/main/java/develop/MyKafkaVirtualObject.java
-```
+# Kafka
+You can invoke handlers via Kafka events, by doing the following:
 
-</TabItem>
-<TabItem value="kotlin" label="Kotlin">
+<SubtleStep stepLabel="1" title="Develop an event handler">
 
-```kotlin
-CODE_LOAD::kotlin/src/main/kotlin/develop/MyKafkaVirtualObject.kt
-```
+You can invoke any handler via Kafka events.
+The event payload will be (de)serialized as JSON.
 
-</TabItem>
-</Tabs>
+- When invoking **Virtual Object** handlers via Kafka, the key of the Kafka record will be used to determine the Virtual Object key.
+For each Virtual Object, the events are delivered in the order in which they arrived on the topic partition.
+- When invoking **Service** handlers over Kafka, events are delivered in parallel without ordering guarantees.
 
-<Admonition type="tip" title="Combining RPC and Kafka within a service">
-You can mix methods receiving events with regular RPC methods. For example, assume you're implementing a `Profile` service to handle and store user profiles in Restate, where the key is the profile's unique identifier.
-In the same service, you can implement an RPC `get` method to retrieve the profile, and you can attach an `updateProfile` event handler to consume events from Kafka to update the profile.
+<Admonition type="tip" title="Combining RPC and Kafka within a service/handler">
+    Since you can invoke any handler via Kafka events, a single handler can be invoked both by RPC and via Kafka.
 </Admonition>
 
-For each event key (= Kafka record key), the events are delivered in the order in which they arrived on the topic partition.
+</SubtleStep>
 
-**Handling events within services**
-
-You can also handle events within services. In this case, events are delivered in parallel without ordering guarantees.
-
-## Event metadata
-
-Each event carries within the `ctx.request().headers()` map the following entries:
-
-* `restate.subscription.id`: The subscription identifier, as shown by the [Admin API](../../references/admin-api).
-* `kafka.offset`: The record offset.
-* `kafka.partition`: The record partition.
-* `kafka.timestamp`: The record timestamp.
```

Comment:
This stuff is important to describe, i don't think it should be removed. perhaps it goes in one of the collapsible sections.

Reply:
Yes, maybe you are right. Added it back